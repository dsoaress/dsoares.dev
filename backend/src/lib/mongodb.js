import { MongoClient } from 'mongodb'

const { MONGODB_URI, MONGODB_DB } = process.env

let cached = global.mongo

if (!cached) {
  cached = global.mongo = { conn: null, promise: null }
}

export async function connectToDatabase() {
  if (cached.conn) {
    return cached.conn
  }

  if (!cached.promise) {
    const opts = {
      useNewUrlParser: true,
      useUnifiedTopology: true
    }

    cached.promise = MongoClient.connect(MONGODB_URI, opts).then(client => {
      return {
        client,
        db: client.db(MONGODB_DB)
      }
    })
  }
  cached.conn = await cached.promise
  return cached.conn
}
