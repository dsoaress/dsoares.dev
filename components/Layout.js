import Header from '@/components/Header'
import Footer from '@/components/Footer'

export default function Layout({ children }) {
  return (
    <>
      <Header />
      <main className="max-w-screen-lg px-8 mx-auto">{children}</main>
      <Footer />
    </>
  )
}
