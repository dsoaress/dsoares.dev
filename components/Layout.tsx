import styled from 'styled-components'
import media from 'styled-media-query'

export const Layout = styled.main`
  display: flex;
  flex-direction: column;
  height: 100vh;

  ${media.greaterThan('medium')`
    flex-direction: row;
  `}
`
