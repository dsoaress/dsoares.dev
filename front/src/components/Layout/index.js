import React from 'react'

import Header from '../Header'
import Footer from '../Footer'
import GlobalStyles from './global'
import * as S from './styled'

const Layout = ({ children }) => (
  <S.Wrapper>
    <GlobalStyles />
    <Header />
    <S.Main>{children}</S.Main>
    <Footer />
  </S.Wrapper>
)

export default Layout
