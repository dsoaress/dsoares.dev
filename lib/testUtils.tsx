import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles/theme'

const Providers = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
