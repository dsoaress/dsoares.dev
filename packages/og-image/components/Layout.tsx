export default function Layout({
  firstName,
  lastName,
  avatar,
  github,
  title,
  url
}) {
  function getFontSize(length) {
    if (length > 32) {
      return `text-7xl`
    }

    return `text-9xl`
  }

  return (
    <div className="relative flex flex-col justify-between p-16 w-[1200px] h-[630px] font-black">
      <div className="max-w-screen-lg">
        <h1 className={getFontSize(title.length)}>{title}</h1>
      </div>
      <div className="flex items-center space-x-6">
        <img
          src={avatar}
          alt={firstName + ' ' + lastName}
          className="flex-none w-32 h-32 border-4 rounded-full border-neutral-200"
        />
        <div>
          <p className="mb-1 text-3xl ">
            {firstName}
            <span className="text-primary-500">{lastName}</span>
          </p>
          <p className="text-2xl">{url}</p>
          <p className="text-2xl text-secondary-400">{github}</p>
        </div>
      </div>
    </div>
  )
}
