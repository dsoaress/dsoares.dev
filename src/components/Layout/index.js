import React from "react"
import PropTypes from "prop-types"

import GlobalStyled from "./global"
import * as S from "./styled"

import Header from "../Header"
import Footer from "../Footer"

const Layout = ({ children }) => {
  return (
    <S.LayoutWrapper>
      <GlobalStyled />
      <Header />
      <S.MainWrapper>{children}</S.MainWrapper>
      <Footer />
    </S.LayoutWrapper>
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
