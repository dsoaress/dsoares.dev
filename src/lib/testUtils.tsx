import { render } from '@testing-library/react'
import { ComponentType, ReactElement, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles/theme'

type ProvidersProps = {
  children?: ReactNode
}

const Providers = ({ children }: ProvidersProps) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
