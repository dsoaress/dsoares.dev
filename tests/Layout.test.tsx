import Layout from '@/components/Layout'
import { render, screen } from '@/lib/testUtils'

describe('<Layout />', () => {
  it('Renders with children', () => {
    render(<Layout>Children</Layout>)

    expect(screen.getByText('Children')).toBeInTheDocument()
    expect(screen.getByText('Children')).toMatchSnapshot()
  })
})
