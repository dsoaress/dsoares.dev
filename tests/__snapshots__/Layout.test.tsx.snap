// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Renders with children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<main
  class="c0"
>
  Children
</main>
`;
